airbyte-integrations/connectors/destination-pgvector/pyproject.toml

```diff
@@ -0,0 +1,86 @@
+[build-system]
+requires = [ "poetry-core>=1.0.0",]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "airbyte-destination-pgvector"
+version = "0.1.0"
+description = "Airbyte destination implementation for PGVector."
+authors = ["Airbyte <contact@airbyte.io>"]
+license = "MIT"
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/destinations/pgvector"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+
+[[tool.poetry.packages]]
+include = "destination_pgvector"
+
+[tool.poetry.dependencies.airbyte-cdk]
+# Version cnstrained by PyAirbyte (`airbyte`) version
+version = ">=1.0"
+extras = ["vector-db-based"]
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+
+airbyte = "^0.12.0"
```

Comment:
I'm just curious — why do we need both the `airbyte` and `airbyte-cdk` packages to build a destination?
If there are missing parts, should we move components or modules from `airbyte` to `airbyte-cdk`?

Reply:
airbyte-integrations/connectors/destination-pgvector/destination_pgvector/pgvector_processor.py
```
from airbyte._processors.file.jsonl import JsonlWriter
from airbyte.secrets import SecretString
````

airbyte-integrations/connectors/destination-pgvector/destination_pgvector/common/sql/sql_processor.py

```
from airbyte._util.name_normalizers import LowerCaseNormalizer
from airbyte.constants import AB_EXTRACTED_AT_COLUMN, AB_META_COLUMN, AB_RAW_ID_COLUMN, DEBUG_MODE
from airbyte.progress import progress
from airbyte.strategies import WriteStrategy
from airbyte.types import SQLTypeConverter
    from airbyte._batch_handles import BatchHandle
    from airbyte._processors.file.base import FileWriterBase
    from airbyte.secrets.base import SecretString
```

Reply:
I could move AirbyteRecordMessage at least.
from airbyte_cdk.models import AirbyteRecordMessage

Reply:
The goal would be to move the SQLProcessor logic (and other classes/functions) into the CDK. But that will take a bit more time, I think.

Reply:
I'm asking because I have concerns about circular import of pyairbyte.

Should we _incorporate_ airbyte inside CDK before publishing the connector?